### PR TITLE
MATX-232 : Add explicit matrix binding workaround for XML->OGSFX lack of value transfer

### PR DESCRIPTION
**Pull request by bernardkwok**
Fixes #MATX-232
Workaround to always bind matrices since XML -> OGSFX does not set values properly.

### CODE REVIEW

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp [hunk 1]
```diff
@@ -323,50 +323,87 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
...
+                    for (unsigned int j = 0; j < 4; j++)
+                    {
+                        mayaValue[i][j] = matrix44[i][j];
+                    }
                 }
-                samplerDescription.addressU = MSamplerState::TextureAddress::kTexWrap;
-                if (samplingProperties.uaddressMode != mx::ImageSamplingProperties::AddressMode::UNSPECIFIED)
+                status = shader.setParameter(resolvedName, mayaValue);
+            }
+            if (mtxValue->isA<mx::Matrix33>())
```
Comment (niklasharrysson) [inline source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp]: Could be an `else if` statement.

File: source/MaterialXContrib/MaterialXNode/MaterialXData.cpp [hunk 2]
```diff
@@ -237,4 +237,9 @@ mx::ImageSamplingProperties MaterialXData::getImageSamplngProperties(const std::
 
     }
     return samplingProperties;
-}
\ No newline at end of file
+}
+
+std::string MaterialXData::getMatrix4Name(const std::string& matrix3Name) const
```
Comment (ppenenko) [inline source/MaterialXContrib/MaterialXNode/MaterialXData.cpp]:
```suggestion
static std::string MaterialXData::getMatrix4Name(const std::string& matrix3Name)
```

File: source/MaterialXContrib/MaterialXNode/MaterialXData.h [hunk 3]
```diff
@@ -69,6 +69,10 @@ class MaterialXData
     /// Get image sampling properties for a given file parameter
     mx::ImageSamplingProperties getImageSamplngProperties(const std::string& fileParameterName) const;
 
+    /// OGS does not support matrix3. As such the matrix4 parameter name is computed from the matrix3 name.
+    /// This utility performs this computation.
+    std::string getMatrix4Name(const std::string& matrix3Name) const;
```
Comment (ppenenko) [inline source/MaterialXContrib/MaterialXNode/MaterialXData.h]:
```suggestion
    static std::string getMatrix4Name(const std::string& matrix3Name);
```

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp [hunk 4]
```diff
@@ -323,50 +323,87 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
...
+                for (unsigned int i = 0; i < 3; i++)
                 {
-                    samplerDescription.addressU = addressModes[static_cast<int>(samplingProperties.uaddressMode)];
+                    for (unsigned int j = 0; j < 3; j++)
+                    {
+                        mayaValue[i][j] = matrix33[i][j];
+                    }
                 }
+                // Note: the parameter exposed uses a derived matrix44 name.
+                std::string matrix4Name = materialXData->getMatrix4Name(resolvedName.asChar());
```
Comment (ppenenko) [inline source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp]:
```suggestion
                std::string matrix4Name = MaterialXData::getMatrix4Name(resolvedName.asChar());
```

**Pull request merged**